server/models/executor.js

```diff
@@ -1,14 +1,53 @@
+var assert = require('assert');
 var async = require('async');
+var debug = require('debug')('strong-mesh-models:service-manager');
+var instModelWatcher = require('../../lib/helper').instModelWatcher;
+var shouldWatch = require('../../lib/helper').shouldWatch;
 
 module.exports = function(Executor) {
 
   // Refer to server-service.js for description of instance vs currentInstance
   // vs data.
 
+  var name = 'executor';
+
   Executor.observe('after save', function(ctx, next) {
     var serviceManager = Executor.app.serviceManager;
-    var instance = ctx.instance || ctx.currentInstance;
+    debug('------------------ Executor SAVE %j.', ctx.where);
+
+    if (serviceManager._dbWatcher) {
+      if (!shouldWatch(serviceManager, name)) {
+        setImmediate(next);
+        return;
+      }
+      var watcherCtx = {
+        'modelName': name,
+        'watcher': serviceManager._dbWatcher,
+        'saveFun': saveObserver,
+        'saveNext': next,
+        'deleteFun': deleteObserver,
+        'modelInst': serviceManager._meshApp.models.Executor,
+      };
+      instModelWatcher(watcherCtx);
+    }
+    saveObserver(ctx, next);
```

Comment:
won't this cause double invocation of the saveObserver once the model watcher is registered?